vignettes/customization.Rmd

```diff
@@ -0,0 +1,222 @@
+
+## Theming
+
+### Choosing a bootswatch theme
+
+You can choose any [Bootswatch theme](https://bootswatch.com/) that's compatible with the Bootstrap version you use.
+Run either `bslib::bootswatch_themes(3)` or `bslib::bootswatch_themes(4)` to get a list.
+If you were using the paper theme, note that it was renamed [materia](https://blog.bootswatch.com/post/167233472837/announcing-bootswatch-400).
+
+To choose a Bootswatch theme here's what to write in your `pkgdown` configuration.
+
+```yaml
+template:
+  bootstrap: 4
+  params:
+    bootswatch: materia
+```
+
+_Some Bootswatch templates like lux or pulse won't work out of the box because of their tall navbar._
+_Add [CSS rules to prevent overlap with other elements](https://getbootstrap.com/docs/4.6/components/navbar/#placement) like [padding to the body](https://stackoverflow.com/questions/10336194/top-nav-bar-blocking-top-content-of-the-page)._
+
+Then, when using Bootstrap 4,
+you can use any variables [documented in bslib](https://rstudio.github.io/bslib/articles/bs4-variables.html) for customizing your theme!
+
+```yaml
+template:
+  bootstrap: 4
+  params:
+    bootswatch: materia
+    bslib:
+      bg: "#202123"
+      fg: "#B8BCC2"
+      primary: "#EA80FC"
+      success: "#4F9B29"
+      info: "#28B3ED"
+      warning: "#FD7424"
+      danger: "#F7367E"
+```
+
+By default pkgdown uses an accessible colour scheme for code highlighting... on a light background.
+If you wish to opt out of this colour scheme to e.g. suit a dark Bootswatch theme better, here's what lines to add in your pkgdown configuration file:
+
+```yaml
+template:
+  params:
+    highlightcss: false
+```
+
+### How to craft and preview style tweaks
+
+When crafting a new style you will need to re-build your website quite often.
+How to have a faster workflow?
+
+* Use a simpler package for previewing your style. This might be over the top for a single package but if you are working on a template package, the template package can contain minimal vignettes, reference pages for testing!
+* Do not rebuild your package every time: e.g. only re-build the home page via `build_home()` to get a preview of the home sidebar, or of the footer and navbar.
+* If you edit CSS, you can run `init_site()` to have them copied to your website folder, and then refresh the website in your browser.
+
+## More assets
+
+If you need to store more CSS rules, some JS code, an SVG used by your CSS, etc.
+
+* Store the files under `inst/pkgdown/assets/BS4` (if you're using Bootstrap 4) e.g. `inst/pkgdown/assets/BS4/cool-styling.css`.
+* They will be copied to the root of your website folder by pkgdown (by `init_site()`).
+* For JS and CSS files, reference them in a `inst/pkgdown/after-head.html` template (see the section "Other layout changes"), e.g.
+
+```html
+<link href="{{#site}}{{root}}{{/site}}cool-styling.css" rel="stylesheet">
+```
+
+## Layout
```

Comment:
I feel like this should come before "more assets" since it'll be more common.

Reply:
I moved it and then moved it back... because the section about theming mentions extra.css right at the end.